rust/doc/api.md

```diff
@@ -0,0 +1,55 @@
+# API Documentation
+
+This Api describes the interface for communicating with the new Rust Scanner.
+It is meant to be used by clients to control scans and manage its data. The core feature is the ability to start networks scans with customized settings to detect vulnerable systems within a network.
+The specification of the interface is defined with the OpenAPI 3.1 standard and is contained within the provided [openapi.yml](openapi.yml) file.
+This way the specification is highly standardized with many tools to generate a documentation for any use.
+A list of tools to use can be found [here](https://openapi.tools/).
+
+The reason for this document is the lack of some specialized features regarding this API, that cannot be specified within the OpenAPI document.
+This features contain the authentication system used for accessing the Scanner API and some special handling regarding meta information contained in the header of a response.
+
+## Authentication
+
+In general the API supports 3 kind of authentication methods:
+- No authentication
+- API Key
+- Certificates
+The different authentication modes are set within a configuration file or via the argument list, when starting the server.
+
+The authentication is required for each request except for a GET request on the root path.
+
+### No Authentication
+
+This mode is not meant to be used in production, but rather for testing and development.
+With this mode no authentication is needed in order to communicate with the Scanner.
+
+### API Key
+
+A API key is a token that client provides when making API requests and are used to authenticate the user.
+The key must be sent within the header.
+The name of the header field containing the API key should be `SCANNER-API-KEY`.
+
+### Certificates
+
+The last option for authentication is the use of [X.509 based certificates](https://en.wikipedia.org/wiki/X.509).
+The certificate will in this case represent the identity of a client and only clients with a registered certificate is granted access to the API.
+In contrast to the API keys this can be used as a bidirectional authentication.
+This means the client can authenticate the server first, then the server authenticate the client.
+
+More details about this authentication method follows with its implementation.
+
+## API Response Header
+
+The API Response Header contains additional information, which can be used by the client to verify the used versions of the scanner components, as well as the version of the API, so it can do the appropriate calls.
+The header will contain the following fields:
+- `api-version` The version of the API
+- `scanner-version` The version of the Scan manager
+- `openvas-version` The version of the OpenVAS scanner
+- `notus-version` The version of the Notus scanner
```

Comment:
do we want to expose openvas-version, notus-version and scanner-version?

Reply:
My thought on this was to get information about what the scanner is actual capable of. E.g. if the notus scanner is not available for this scanner, this can be took into account.
But also maybe such things should be handled within the scanner internally, so the client does not need such information. What I mean of this, in the end the client does not care if either OpenVAS or Notus handles VTs.